src/vm/node_modules/VM.js

```diff
@@ -11009,7 +11026,11 @@ exports.reprovision = function (uuid, payload, options, callback)
             });
         }, function (cb) {
             // unmount <zonepath>/cores so dataset is not busy
-            zfs(['umount', vmobj.zonepath + '/cores'], log,
+            // unmounting using the mountpoint occasionally fails with:
+            //      "cannot unmount '/zones/<vm_uuid>/cores': not a mountpoint"
+            // so use the dataset name instead of the mountpoint which seems to
+            // solve the issue
+            zfs(['unmount', vmobj.zpool + '/cores/' + uuid], log,
```

Comment:
Another thing about this, is the "not a mountpoint" message is explicitly handled immediately below in the next stanza, but the error handling is not update.

This will turn a non-fatal error into a fatal error.